multimodal/src/autogluon/multimodal/utils.py

```diff
@@ -2781,3 +2782,40 @@ def move_to_device(obj: Union[torch.Tensor, nn.Module, Dict, List], device: torc
             f"Make sure the object is one of these: a Pytorch tensor, a Pytorch module, "
             f"a dict or list of tensors or modules."
         )
+
+
+def infer_batch(batch: Dict, model: nn.Module, precision: Union[str, int], loss_func: _Loss):
+    """
+    Perform inference for a batch.
+
+    Parameters
+    ----------
+    batch
+        The batch data.
+    model
+        A Pytorch model.
+    precision
+        The desired precision used in inference.
+    loss_func
+        The loss function used in training the model.
+
+    Returns
+    -------
+    Model output.
+    """
+    precision = infer_precision(num_gpus=1, precision=precision, as_torch=True)
+    device_type = "cuda" if torch.cuda.is_available() else "cpu"
+    device = torch.device(device_type)
+    batch_size = len(batch[next(iter(batch))])
+    if 1 < torch.cuda.device_count() <= batch_size:
+        model = nn.DataParallel(model)
+    model.to(device).eval()
+    batch = move_to_device(batch, device=device)
+    with torch.autocast(device_type=device_type, dtype=precision):
+        with torch.no_grad():
+            output = model(batch)[model.prefix]
+
+    if isinstance(loss_func, nn.BCEWithLogitsLoss):
```

Comment:
Should we put it under `torch.no_grad()?`

Reply:
Done.

Reply:
BTW, I am thinking about a form way to handle this kind of post-processing like `nn.BCEWithLogitsLoss` . The current way is kind of a hack and have several repeated code snippets, which is not easy to maintain.